### ISSUE TITLE
Gradient on Embedding padding vector is nonzero after forward on inputs longer than 768 on GPU

### ISSUE THREAD

**Issue by michaelcandido** [pytorch/pytorch#28716]
## 🐛 Bug

According to torch.nn documentation, the Embedding module allows assigning a padding_idx to one embedding vector. This vector (assuming default initialization) will always be 0's and have 0 gradient. While working with a long sequence model (32 x 1000 inputs), I noticed the embedding vector for the padding index was becoming nonzero during training. To debug, I created an isolated repro of the issue (embed + sum + backward) and found that the gradient on the padding vector is zero as expected for a 1 x 768 input and anything smaller, but it suddenly becomes nonzero for a 1 x 769 input (or larger). Additionally, the bug only happens when the embedding and input tensor are on the GPU.

## To Reproduce

emb = nn.Embedding(num_embeddings=10, embedding_dim=8, padding_idx=0, sparse=False).cuda()
x = torch.zeros(769, dtype=torch.int64).unsqueeze(0).cuda()
x = emb(x)
x = x.sum()
x.backward()
print(emb.weight.grad[0])

Output: tensor([768., 768., 768., 768., 768., 768., 768., 768.], device='cuda:0')

If you change the torch.zeros dim to 768, the gradient is 0. Likewise if you remove the calls to cuda(), the gradient is 0. I ran this repro in colab. Here's a runnable notebook link: https://colab.research.google.com/gist/michaelcandido/2915eae4d7c07e5527ea9eccb041db07/pytorch-embedding-padding-bug.ipynb

## Expected behavior

emb.weight.grad[0] should be 0, so the padding vector remains nonzero during training

Expected: tensor([0., 0., 0., 0., 0., 0., 0., 0.], device='cuda:0')

## Environment

Google Colab

Collecting environment information...
PyTorch version: 1.3.0+cu100
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.12.0

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Tesla K80
Nvidia driver version: 418.67
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.3

Versions of relevant libraries:
[pip3] numpy==1.17.3
[pip3] torch==1.3.0+cu100
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.3.1
[pip3] torchvision==0.4.1+cu100
[conda] Could not collect

**Comment by YilunZhou** [comment]
looks like it is already fixed? I can't reproduce the error on the colab script anymore.

**Comment by jbschlosser** [comment]
Closing since this is no longer reproducible.
